src/electron/views/common/body-class-modifier/body-class-modifier.tsx

```diff
@@ -1,12 +1,13 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import { PlatformInfo } from 'electron/window-management/platform-info';
+import { ClassAssigner } from 'electron/views/common/body-class-modifier/class-assigner';
+import { uniq } from 'lodash';
 import { css } from 'office-ui-fabric-react';
 import * as React from 'react';
 import Helmet from 'react-helmet';
 
 export type BodyClassModifierDeps = {
-    platformInfo: PlatformInfo;
+    classAssigners: ClassAssigner[];
```

Comment:
I worry that encapsulating the information that determines className behind a function call like this may lead to issues with React not propogating updates to this component like we'd expect; it doesn't matter here when it's just `platformInfo` since it doesn't change over time, but it might when we add high contrast mode and need React to understand to rerender this component when the high contrast mode settings change.

I wonder if it would make more sense to re-use the existing `Theme` component that the extension uses for the high contrast body classes, give it an `extraClassNames` prop, and pass it `calculatePlatformClassNames(platformInfo)` from the unified initializer similar to what you're doing to pass the classAssigners in this PR.

Reply:
So... In a different branch ([this one](https://github.com/smoralesd/accessibility-insights-web/tree/high-contrast-theme-class-assigner)), I'm introducing `ThemeClassAssigner`. I converted `BodyClassAssigner` so it uses `withStoreSubscription`. This ensure the updates you're worried about happen when the user enable/disable high contrast mode (it's working properly on the branch I mention). I'm not 100% convinced on the design though. The fact that `BodyClassAssigner` "listen" to store changes to trigger the class assigners but doesn't explicitly pass the state to them (nor it uses the state directly) seems odd to me. I'm open to suggestions.

Reply:
I agree that it doesn't feel right for the `BodyClassModifier` to appear to listen to state changes but not directly use the state; I think it would be best to make the coupling more obvious. Maybe instead of `assign()` and `update()`, just do `assign(userConfig: UserConfigurationStoreData)`, and make `ThemeClassAssigner` stateless?